### PR TITLE
#8614: rename remaining instances of recipe in the page editor

### LINKED ISSUE #8616: Support Google Sheets Playwright tests work on CI
Context
---
- https://github.com/pixiebrix/pixiebrix-extension/issues/8058

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Part of #8614

## Future Work

- Redux Persist migration to change `BaseFormState` to change `recipe` key to `mod`

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

**Comment by grahamlangford** [comment]
The change to get/setPageState blew up the PR more than I realized. I can revert that change to reduce the scope of the PR if needed

File: src/bricks/effects/assignModVariable.ts [hunk 1]
```diff
@@ -102,8 +102,8 @@ class AssignModVariable extends EffectABC {
       namespace: StateNamespaces.MOD,
       data: { [variableName]: value },
       mergeStrategy: MergeStrategies.SHALLOW,
-      extensionId,
-      blueprintId,
+      modComponentId: extensionId,
+      modId: blueprintId,
```
Comment (mnholtz) [inline src/bricks/effects/assignModVariable.ts:106]: nit: opportunity to rename the local variables as well

File: src/pageEditor/starterBricks/sidebar.ts [hunk 2]
```diff
@@ -129,14 +129,14 @@ function asDraftModComponent(
...
+  const extensionPoint = await lookupStarterBrick<
     SidebarDefinition,
     SidebarConfig,
     "actionPanel"
   >(config, "actionPanel");
 
-  const base = baseFromExtension(config, extensionPoint.definition.type);
-  const extension = await extensionWithNormalizedPipeline(
+  const base = baseFromModComponent(config, extensionPoint.definition.type);
+  const extension = await modComponentWithNormalizedPipeline(
```
Comment (mnholtz) [inline src/pageEditor/starterBricks/sidebar.ts:139]: nit: opportunity to rename these local vars, and the one on line 132

File: src/pageEditor/starterBricks/tour.ts [hunk 3]
```diff
@@ -115,16 +115,16 @@ function asDraftModComponent(tourFormState: TourFormState): DraftModComponent {
...
     TourConfig,
     "tour"
   >(config, "tour");
 
   const { reader } = extensionPoint.definition;
 
-  const base = baseFromExtension(config, extensionPoint.definition.type);
-  const extension = await extensionWithNormalizedPipeline(
+  const base = baseFromModComponent(config, extensionPoint.definition.type);
+  const extension = await modComponentWithNormalizedPipeline(
```
Comment (mnholtz) [inline src/pageEditor/starterBricks/tour.ts:127]: nit: opportunity to rename local car

File: src/pageEditor/starterBricks/trigger.ts [hunk 4]
```diff
@@ -174,8 +174,8 @@ async function fromExtension(
     customEvent,
   } = extensionPoint.definition;
 
-  const base = baseFromExtension(config, extensionPoint.definition.type);
-  const extension = await extensionWithNormalizedPipeline(
+  const base = baseFromModComponent(config, extensionPoint.definition.type);
+  const extension = await modComponentWithNormalizedPipeline(
```
Comment (mnholtz) [inline src/pageEditor/starterBricks/trigger.ts:178]: nit: opportunity to rename local vars

File: src/pageEditor/tabs/editTab/dataPanel/tabs/PageStateTab.test.tsx [hunk 5]
```diff
@@ -53,13 +53,13 @@ describe("PageStateTab", () => {
     return utils;
   }
 
-  test("it renders with orphan extension", async () => {
+  test("it renders with orphan mod component", async () => {
```
Comment (mnholtz) [inline src/pageEditor/tabs/editTab/dataPanel/tabs/PageStateTab.test.tsx:56]:
```suggestion
  test("it renders with standalone mod component", async () => {
```

File: src/pageEditor/tabs/editTab/dataPanel/tabs/__snapshots__/PageStateTab.test.tsx.snap [hunk 6]
```diff
@@ -143,20 +143,72 @@ exports[`PageStateTab it renders with orphan extension 1`] = `
               </ul>
             </li>
             <li
-              style="padding-top: 0.25em; padding-right: 0px; margin-left: 0.875em; word-wrap: break-word; padding-left: 1.25em; text-indent: -0.5em; word-break: break-all; white-space: pre-wrap;"
+              style="position: relative; padding-top: 0.25em; margin-left: 0px; padding-left: 0px;"
```
Comment (mnholtz) [inline src/pageEditor/tabs/editTab/dataPanel/tabs/__snapshots__/PageStateTab.test.tsx.snap:146]: What caused this style to change? 🤔

Comment (mnholtz) [reply src/pageEditor/tabs/editTab/dataPanel/tabs/__snapshots__/PageStateTab.test.tsx.snap:146]: Oh, nevermind. It looks like the snapshots got swapped. Maybe they're alphabetized

File: src/platform/state/stateController.test.ts [hunk 7]
```diff
@@ -56,16 +56,16 @@ describe("pageState", () => {
         asyncState: { isFetching: false, data: "foo", currentData: "foo" },
       },
       mergeStrategy: MergeStrategies.DEEP,
-      extensionId: uuidv4(),
-      blueprintId,
+      modComponentId: uuidv4(),
+      modId: blueprintId,
```
Comment (mnholtz) [inline src/platform/state/stateController.test.ts:60]: nit: opportunity to rename local variable

File: src/platform/state/stateController.ts [hunk 8]
```diff
@@ -123,8 +123,8 @@ export function setState({
       previous,
```
Comment (mnholtz) [inline src/platform/state/stateController.ts:123]: I just realized... is this func supposed to be called dispatch**State**ChangeEventOnChange?

File: src/platform/state/stateController.ts [hunk 9]
```diff
@@ -123,8 +123,8 @@ export function setState({
...
       next,
       namespace,
-      extensionId,
-      blueprintId,
+      extensionId: modComponentId,
+      blueprintId: modId,
```
Comment (mnholtz) [inline src/platform/state/stateController.ts:127]: nit: option to change function signature if you end up fixing the typo on the function name

File: src/platform/state/stateController.ts [hunk 10]
```diff
@@ -138,18 +138,21 @@ export function setState({
...
       notifyOnChange(previous, next);
       return next;
     }
 
     case StateNamespaces.PRIVATE: {
-      assertNotNullish(extensionId, "Invalid context: extensionId not found");
-      const previous = privateState.get(extensionId) ?? {};
+      assertNotNullish(
+        modComponentId,
+        "Invalid context: extensionId not found",
```
Comment (mnholtz) [inline src/platform/state/stateController.ts:151]:
```suggestion
        "Invalid context: modComponentId not found",
```

File: src/platform/state/stateController.ts [hunk 11]
```diff
@@ -179,12 +182,15 @@ export function getState({
...
-      return modState.get(blueprintId) ?? {};
+      return modState.get(modId) ?? {};
     }
 
     case StateNamespaces.PRIVATE: {
-      assertNotNullish(extensionId, "Invalid context: extensionId not found");
-      return privateState.get(extensionId) ?? {};
+      assertNotNullish(
+        modComponentId,
+        "Invalid context: extensionId not found",
```
Comment (mnholtz) [inline src/platform/state/stateController.ts:191]:
```suggestion
        "Invalid context: modComponentId not found",
```

**Review by mnholtz — APPROVED** [review]

**Pull request merged**